### DISCUSSION TITLE
How to set up rust-analyzer for linux kernel module development?

### DISCUSSION THREAD

**Discussion by Tunghohin** [rust-lang/rust-analyzer#15850] [Q&A] [answered]
It seems that rust-analyzer does not provide static syntax checking and code completion in linux/sample/rust.

**Comment by lnicola** [comment]
Did you try [`scripts/generate_rust_analyzer.py`](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/scripts/generate_rust_analyzer.py?h=v6.6)?

**Reply by lnicola** [reply, marked as answer]
Actually, that might be https://docs.kernel.org/rust/quick-start.html#rust-analyzer.

**Reply by Tunghohin** [reply]
Thank you very much. Forgive my carelessness.
